sdk/cosmos/azure_data_cosmos/src/retry_policies/client_retry_policy.rs

```diff
@@ -164,10 +164,15 @@ impl ClientRetryPolicy {
                 .resolve_service_endpoint(request),
         );
 
+        tracing::trace!(
+            ?self.location_endpoint,
+            "routed request to endpoint"
+        );
+
         if let Some(ref endpoint) = self.location_endpoint {
-            request.request_context.route_to_location_endpoint(
-                request.resource_link.url(&Url::parse(endpoint).unwrap()),
-            );
+            request
+                .request_context
+                .route_to_location_endpoint(endpoint.clone());
```

Comment:
@kundadebdatta see below for context. This is where I noticed that we're sometimes setting the location endpoint to include the path to the resource.